### PR TITLE
Fix uv issue on windows

### PR DESCRIPTION
**Pull request by khoaguin**
image

### CODE REVIEW

**Comment by Smartappli** [comment]
https://github.com/Smartappli/PySyft/commit/43691aaa5ca0900064816d83f214901b0aced45b

File: .github/workflows/pr-tests-syft.yml
Comment (Smartappli) [inline .github/workflows/pr-tests-syft.yml:1]: python -m pip install --upgrade pip==24.1

python -m pip install uv==0.2.17

Comment (khoaguin) [reply .github/workflows/pr-tests-syft.yml:1]: updated accordingly, thanks @Smartappli

File: .github/workflows/cd-docs.yml [hunk 1]
```diff
@@ -27,7 +27,7 @@ jobs:
 
       - name: Install tox
         run: |
-          pip install --upgrade pip==24.0 uv==0.2.13 tox tox-uv==1.9.0
+          pip install --upgrade pip==24.1 uv==0.2.17 tox tox-uv==1.9.0
```
Comment (Smartappli) [inline .github/workflows/cd-docs.yml]: pip install --upgrade pip==24.1.1 uv==0.2.17
uv --version
uv pip install tox tox-uv==1.9.0

Comment (Smartappli) [reply .github/workflows/cd-docs.yml]: might as well use uv since it's installed

File: .github/workflows/cd-syft-dev.yml [hunk 2]
```diff
@@ -164,7 +164,7 @@ jobs:
           helm version
 
           # install tox
-          python -m pip install --upgrade pip==24.0
+          python -m pip install --upgrade pip==24.1
```
Comment (Smartappli) [inline .github/workflows/cd-syft-dev.yml]: python -m pip install --upgrade pip==24.1.1
          uv --version
          uv pip install tox

File: .github/workflows/cd-syft.yml [hunk 3]
```diff
@@ -134,7 +134,7 @@ jobs:
 
       - name: Install dependencies
         run: |
-          pip install --upgrade pip==24.0 uv==0.2.13 bump2version tox tox-uv==1.9.0
+          pip install --upgrade pip==24.1 uv==0.2.17 bump2version tox tox-uv==1.9.0
```
Comment (Smartappli) [inline .github/workflows/cd-syft.yml]: pip install --upgrade pip==24.1.1 uv==0.2.17
uv --version
uv pip install  bump2version tox tox-uv==1.9.0

File: .github/workflows/cd-syftcli.yml [hunk 4]
```diff
@@ -65,7 +65,7 @@ jobs:
       - name: Install dependencies
         if: ${{steps.get-hashes.outputs.current_hash != steps.get-hashes.outputs.previous_hash }}
         run: |
-          python -m pip install --upgrade pip==24.0
+          python -m pip install --upgrade pip==24.1
```
Comment (Smartappli) [inline .github/workflows/cd-syftcli.yml]: python -m pip install --upgrade pip==24.1.1 uv==0.2.17
uv --version
uv pip install --upgrade tox setuptools wheel twine bump2version PyYAML

File: .github/workflows/e2e-tests-notebook.yml [hunk 5]
```diff
@@ -63,7 +63,7 @@ jobs:
 
       - name: Install Deps
         run: |
-          pip install --upgrade pip==24.0 uv==0.2.13 tox tox-uv==1.9.0
+          pip install --upgrade pip==24.1 uv==0.2.17 tox tox-uv==1.9.0
```
Comment (Smartappli) [inline .github/workflows/e2e-tests-notebook.yml]: pip install --upgrade pip==24.1.1 uv==0.2.17
uv --version
uv pip install tox tox-uv==1.9.0

File: .github/workflows/pr-tests-frontend.yml [hunk 6]
```diff
@@ -46,7 +46,7 @@ jobs:
       - name: Upgrade pip
         if: steps.changes.outputs.frontend == 'true'
         run: |
-          pip install --upgrade pip==24.0 uv==0.2.13
+          pip install --upgrade pip==24.1 uv==0.2.17
```
Comment (Smartappli) [inline .github/workflows/pr-tests-frontend.yml]: L75: uv pip install --upgrade tox tox-uv==1.9.0

File: .github/workflows/pr-tests-helm-lint.yml [hunk 7]
```diff
@@ -33,7 +33,7 @@ jobs:
           brew install kube-linter FairwindsOps/tap/polaris
 
           # Install python deps
-          pip install --upgrade pip==24.0
+          pip install --upgrade pip==24.1
```
Comment (Smartappli) [inline .github/workflows/pr-tests-helm-lint.yml]: pip install --upgrade pip==24.1.1 uv==0.2.17
uv --version
uv pip install tox

File: .github/workflows/pr-tests-linting.yml [hunk 8]
```diff
@@ -29,7 +29,7 @@ jobs:
 
       - name: Install pip packages
         run: |
-          pip install --upgrade pip==24.0 uv==0.2.13
+          pip install --upgrade pip==24.1 uv==0.2.17
```
Comment (Smartappli) [inline .github/workflows/pr-tests-linting.yml]: L52: uv pip install --upgrade tox tox-uv==1.9.0

File: .github/workflows/pr-tests-stack.yml [hunk 9]
```diff
@@ -52,7 +52,7 @@ jobs:
       - name: Upgrade pip
         if: steps.changes.outputs.stack == 'true'
         run: |
-          pip install --upgrade pip==24.0 uv==0.2.13
+          pip install --upgrade pip==24.1 uv==0.2.17
```
Comment (Smartappli) [inline .github/workflows/pr-tests-stack.yml]: L77: uv pip install --upgrade tox tox-uv==1.9.0

File: .github/workflows/pr-tests-stack.yml [hunk 10]
```diff
@@ -113,7 +113,7 @@ jobs:
...
```
Comment (Smartappli) [inline .github/workflows/pr-tests-stack.yml]: L138; uv pip install --upgrade tox tox-uv==1.9.0

File: .github/workflows/pr-tests-stack.yml [hunk 11]
```diff
@@ -200,7 +200,7 @@ jobs:
...
```
Comment (Smartappli) [inline .github/workflows/pr-tests-stack.yml]: L225: uv pip install --upgrade tox tox-uv==1.9.0

File: .github/workflows/pr-tests-stack.yml [hunk 12]
```diff
@@ -353,7 +353,7 @@ jobs:
...
```
Comment (Smartappli) [inline .github/workflows/pr-tests-stack.yml]: L378; uv pip install --upgrade tox tox-uv==1.9.0

File: .github/workflows/pr-tests-syft.yml [hunk 13]
```diff
@@ -65,12 +65,12 @@ jobs:
...
         if: steps.changes.outputs.syft == 'true'
         run: |
-          python -m pip install --upgrade pip==24.0
+          python -m pip install --upgrade pip==24.1
 
       - name: Install uv
         if: steps.changes.outputs.syft == 'true'
         run: |
-          python -m pip install uv==0.2.13
+          python -m pip install uv==0.2.17
```
Comment (Smartappli) [inline .github/workflows/pr-tests-syft.yml]: L102: uv  pip install --upgrade tox tox-uv==1.9.0

File: .github/workflows/pr-tests-syft.yml [hunk 14]
```diff
@@ -249,7 +249,7 @@ jobs:
       - name: Upgrade pip
         if: steps.changes.outputs.stack == 'true' || steps.changes.outputs.notebooks == 'true'
         run: |
-          pip install --upgrade pip==24.0 uv==0.2.13
+          pip install --upgrade pip==24.1 uv==0.2.17
```
Comment (Smartappli) [inline .github/workflows/pr-tests-syft.yml]: L274: uv pip install --upgrade tox tox-uv==1.9.0

File: .github/workflows/pr-tests-syft.yml [hunk 15]
```diff
@@ -350,7 +350,7 @@ jobs:
       - name: Upgrade pip
         if: steps.changes.outputs.syft == 'true'
         run: |
-          pip install --upgrade pip==24.0 uv==0.2.13
+          pip install --upgrade pip==24.1 uv==0.2.17
```
Comment (Smartappli) [inline .github/workflows/pr-tests-syft.yml]: L375: uv pip install --upgrade tox tox-uv==1.9.0

File: .github/workflows/cd-syft.yml [hunk 16]
```diff
@@ -386,7 +386,7 @@ jobs:
           python-version: "3.12"
       - name: Install dependencies
         run: |
-          pip install --upgrade pip==24.0 uv==0.2.13 tox tox-uv==1.9.0 setuptools wheel twine bump2version PyYAML
+          pip install --upgrade pip==24.1 uv==0.2.17 tox tox-uv==1.9.0 setuptools wheel twine bump2version PyYAML
```
Comment (Smartappli) [inline .github/workflows/cd-syft.yml]: python -m pip install --upgrade pip==24.1.1 uv==0.2.17
uv --version
uv pip install  tox tox-uv==1.9.0 setuptools wheel twine bump2version PyYAML

File: .github/workflows/cd-post-release-tests.yml [hunk 17]
```diff
@@ -193,7 +193,7 @@ jobs:
 
       - name: Install tox and uv
         run: |
-          pip install --upgrade pip==24.0 uv==0.2.13 tox tox-uv==1.9.0 tox-current-env
+          pip install --upgrade pip==24.1 uv==0.2.17 tox tox-uv==1.9.0 tox-current-env
```
Comment (Smartappli) [inline .github/workflows/cd-post-release-tests.yml]: pip install --upgrade pip==24.1.1 uv==0.2.17
uv --version
uv pip install  tox tox-uv==1.9.0 tox-current-env

File: .github/workflows/cd-post-release-tests.yml [hunk 18]
```diff
@@ -132,7 +132,7 @@ jobs:
 
       - name: Install tox
         run: |
-          pip install --upgrade pip==24.0 uv==0.2.13 tox tox-uv==1.9.0
+          pip install --upgrade pip==24.1 uv==0.2.17 tox tox-uv==1.9.0
```
Comment (Smartappli) [inline .github/workflows/cd-post-release-tests.yml]: pip install --upgrade pip==24.1.1 uv==0.2.17
uv --version
uv pip install tox tox-uv==1.9.0

File: .github/workflows/cd-syftcli.yml [hunk 19]
```diff
@@ -121,7 +121,7 @@ jobs:
 
       - name: Install build dependencies for syftcli
         run: |
-          pip install --upgrade pip==24.0
+          pip install --upgrade pip==24.1
```
Comment (Smartappli) [inline .github/workflows/cd-syftcli.yml]: pip install --upgrade pip==24.1.1

File: .github/workflows/pr-tests-frontend.yml (hunk 6 above)
Comment (Smartappli) [inline .github/workflows/pr-tests-frontend.yml]: pip install --upgrade pip==24.1.1 uv==0.2.17

File: .github/workflows/pr-tests-helm-upgrade.yml [hunk 20]
```diff
@@ -37,7 +37,7 @@ jobs:
           brew update
 
           # Install python deps
-          pip install --upgrade pip==24.0
+          pip install --upgrade pip==24.1
```
Comment (Smartappli) [inline .github/workflows/pr-tests-helm-upgrade.yml]: pip install --upgrade pip==24.1 uv==0.2.17
uv --version
uv pip pip install tox

File: .github/workflows/pr-tests-linting.yml (hunk 8 above)
Comment (Smartappli) [inline .github/workflows/pr-tests-linting.yml]: pip install --upgrade pip==24.1.1 uv==0.2.17

File: .github/workflows/pr-tests-stack.yml (hunk 9 above)
Comment (Smartappli) [inline .github/workflows/pr-tests-stack.yml]: pip install --upgrade pip==24.1.1 uv==0.2.17

File: .github/workflows/pr-tests-stack.yml (hunk 10 above)
Comment (Smartappli) [inline .github/workflows/pr-tests-stack.yml]: pip install --upgrade pip==24.1.1 uv==0.2.17

File: .github/workflows/pr-tests-stack.yml (hunk 11 above)
Comment (Smartappli) [inline .github/workflows/pr-tests-stack.yml]: pip install --upgrade pip==24.1.1 uv==0.2.17

File: .github/workflows/pr-tests-stack.yml (hunk 12 above)
Comment (Smartappli) [inline .github/workflows/pr-tests-stack.yml]: pip install --upgrade pip==24.1.1 uv==0.2.17

File: .github/workflows/pr-tests-syft.yml [hunk 21]
```diff
@@ -65,8 +65,14 @@ jobs:
       - name: Upgrade pip
         if: steps.changes.outputs.syft == 'true'
         run: |
-          pip install --upgrade pip==24.0 uv==0.2.13
-          uv --version
+          python -m pip install --upgrade pip==24.1
```
Comment (Smartappli) [inline .github/workflows/pr-tests-syft.yml]: python -m pip install --upgrade pip==24.1.1

File: .github/workflows/pr-tests-syft.yml [hunk 22]
```diff
@@ -160,7 +166,8 @@ jobs:
       - name: Upgrade pip
         if: steps.changes.outputs.syft == 'true'  || steps.changes.outputs.notebooks == 'true'
         run: |
-          pip install --upgrade pip==24.0 uv==0.2.13
+          pip install --upgrade pip==24.1
```
Comment (Smartappli) [inline .github/workflows/pr-tests-syft.yml]: python -m pip install --upgrade pip==24.1.1 uv==0.2.17

File: .github/workflows/pr-tests-syft.yml [hunk 23]
```diff
@@ -242,7 +249,7 @@ jobs:
       - name: Upgrade pip
         if: steps.changes.outputs.stack == 'true' || steps.changes.outputs.notebooks == 'true'
         run: |
-          pip install --upgrade pip==24.0 uv==0.2.13
+          pip install --upgrade pip==24.1 uv==0.2.17
```
Comment (Smartappli) [inline .github/workflows/pr-tests-syft.yml]: pip install --upgrade pip==24.1.1 uv==0.2.17

File: .github/workflows/pr-tests-syft.yml [hunk 24]
```diff
@@ -343,7 +350,7 @@ jobs:
       - name: Upgrade pip
         if: steps.changes.outputs.syft == 'true'
         run: |
-          pip install --upgrade pip==24.0 uv==0.2.13
+          pip install --upgrade pip==24.1 uv==0.2.17
```
Comment (Smartappli) [inline .github/workflows/pr-tests-syft.yml]: pip install --upgrade pip==24.1.1 uv==0.2.17

File: .github/workflows/cd-docs.yml (hunk 1 above)
Comment (yashgorana) [reply .github/workflows/cd-docs.yml]: We use uv solely for making our tox commands faster & not for installing global packages. Hence we want to keep uv & tox global.

**Review by yashgorana — APPROVED** [review]
Great work @khoaguin! This should hopefully fix the nightlies.

**Pull request merged**